src/atframe.jl

```diff
@@ -97,6 +75,11 @@ setanimation!(vis, anim)
 ```
 """
 function atframe(f, animation::Animation, frame::Integer)
-    Cassette.overdub(AnimationCtx(metadata=(animation, frame)), f)
+    push!(animation.visualizer.animation_contexts, AnimationContext(animation, frame))
+    try
+        f()
+    finally
+        pop!(animation.visualizer.animation_contexts)
+    end
     return animation
 end
```

Comment:
I think I finally understood how animations are implemented in MeshCat.jl thanks to these changes.

There is a stack of objects called animation contexts, which are a kind of "helper" object to keep track of which transformations and properties are applied on different objects at different frames. This works well because the methods that apply the transformations and properties are first checking for the existence of a context before applying said transformation or property to the object. I.e., if there is no animation context, we apply the transforms directly to the objects in the visualisation; otherwise, if there is an animation context, it must be because the `settransform`, etc. methods are being called within an `atframe` call, and so we apply the transforms to the last animation context in the stack instead of to the visualiser directly.

Initially, I was confused with the `pop!` instruction of the context. I now understand that we `push!` it temporarily, then apply the "state of the world" as specified in an `atframe` call, and finally we `pop!` it back so that future `settransform` calls are applied directly to the visualiser again rather than to some animation context.

I suppose that there are other ways of achieving this without the need for pushing and popping contexts to a stack, but I guess that doesn't really matter as this is working really well and is a nice way of implementing the logic for conditionally applying `settransform!` and `setprop!` to either an animation context or directly to the visualiser depending on whether there is a context in the stack.  👍